runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -202,36 +230,53 @@ void registerTable(@Nonnull String namespace,
+             *    2.1.1: Each of Existing protobuf file descriptors match that in new file descriptor list
+             *    2.1.2: Existing protobuf file descriptor does not match that in new file descriptor![ SchemaChange!]
+             *
+             *    2.2: Existing protobuf file list does not match, newly opened protobuf list [big Schema Change]
+             *    2.2: Existing protobuf file list matches existing proto files
+             *    2.3: Existing protobuf file's descriptor mismatches with the incoming descriptor
+             */
+            if (TransactionalContext.isInTransaction()) {
+                throw new IllegalThreadStateException("openTable: Called on an existing transaction");
+            }
             try {
                 this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
-                boolean hasSchemaChanged = false;
                 CorfuRecord<TableDescriptors, TableMetadata> oldRecord = this.registryTable.get(tableNameKey);
-                if (oldRecord != null && !oldRecord.getPayload().getFileDescriptorsMap()
-                        .equals(tableDescriptors.getFileDescriptorsMap())) {
-                    hasSchemaChanged = true;
-                    log.warn("registerTable: Schema update detected for table {}${}", namespace, tableName);
-
-                    for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
-                        log.warn("{}", st);
-                    }
-                    log.debug("registerTable: old schema: {}", oldRecord.getPayload().getFileDescriptorsMap());
-                    log.debug("registerTable: new schema: {}", tableDescriptors.getFileDescriptorsMap());
-                }
-                if (hasSchemaChanged) {
+                if (oldRecord == null) { // Case 1 above, new unseen table
                     this.registryTable.put(tableNameKey,
                             new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
-                } else {
-                    this.registryTable.putIfAbsent(tableNameKey,
-                            new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
+                } else { // Case 2 above, re-open of existing table
+                    if (!oldRecord.getPayload().getFileDescriptorsMap().keySet()
+                            .equals(tableDescriptors.getFileDescriptorsMap().keySet())) {
+                        log.warn("Protobuf file list changed for {} table {}. Prev list {}, new list {}",
+                                namespace, tableName, oldRecord.getPayload().getFileDescriptorsMap().keySet(),
+                                tableDescriptors.getFileDescriptorsMap().keySet());
+                        for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
+                            log.debug("{}", st);
+                        }
+                        this.registryTable.put(tableNameKey,
```

Comment:
should this be treated as an error instead?  Is it right to continue in this case?

Reply:
In data-migration this is the normal process, also in build-to-build experiments that different developers do from now until the release, this path is also expected and should not be treated as an error.